app/models/tag.rb

```diff
@@ -1212,8 +1220,18 @@ def self.search(options={})
       query do
         boolean do
           must { string options[:name], default_operator: "AND" } if options[:name].present?
-          must { term '_type', options[:type].downcase } if options[:type].present?
           must { term :canonical, 'T' } if options[:canonical].present?
+
+          if options[:type].present?
+            # To support the tags indexed prior to IndexSubqueue, we want to
+            # find the type either in the tag_type field or the _type field:
+            should { term '_type', options[:type].downcase }
+            should { term :tag_type, options[:type].downcase }
+
+            # The tire gem doesn't natively support :minimum_should_match,
+            # but elasticsearch 0.90 does, so we hack it in.
+            @value[:minimum_should_match] = 1
+          end
```

Comment:
There's no good reason for this whole method to be duplicated, but we can make a note to remove it when we're whacking all the old Tire code.